### ISSUE TITLE
Crash on removing operator inputs 

### ISSUE THREAD

**Issue by hollyhook** [still-scene/t3#176]
1) select input
2) press delete
-> crash

The operator was something in development, thus, it also could be the source of the problem, then please ignore, and close the issue.

However, from the look of it it might be something not really related to my operator.

100% reproducible with my operator, but not with others.

```
System.ArgumentOutOfRangeException
  HResult=0x80131502
  Message=Specified argument was out of the range of valid values. Arg_ParamName_Name
  Source=Microsoft.CodeAnalysis
  StackTrace:
   at Microsoft.CodeAnalysis.SyntaxList`1.get_Item(Int32 index)
   at T3.Gui.Graph.Interaction.NodeOperations.NodeByAttributeIdFinder.VisitFieldDeclaration(FieldDeclarationSyntax node) in C:\dev\repos\t3\t3\Gui\Graph\Interaction\NodeOperations.cs:line 678
   at Microsoft.CodeAnalysis.CSharp.Syntax.FieldDeclarationSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.VisitClassDeclaration(ClassDeclarationSyntax node)
   at Microsoft.CodeAnalysis.CSharp.Syntax.ClassDeclarationSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.VisitNamespaceDeclaration(NamespaceDeclarationSyntax node)
   at Microsoft.CodeAnalysis.CSharp.Syntax.NamespaceDeclarationSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.VisitCompilationUnit(CompilationUnitSyntax node)
   at Microsoft.CodeAnalysis.CSharp.Syntax.CompilationUnitSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxRewriter.Visit(SyntaxNode node)
   at T3.Gui.Graph.Interaction.NodeOperations.RemoveNodesByIdFromTree(Guid[] inputIdsToRemove, SyntaxNode root) in C:\dev\repos\t3\t3\Gui\Graph\Interaction\NodeOperations.cs:line 721
   at T3.Gui.Graph.Interaction.NodeOperations.RemoveInputsFromSymbol(Guid[] inputIdsToRemove, Symbol symbol) in C:\dev\repos\t3\t3\Gui\Graph\Interaction\NodeOperations.cs:line 705
   at T3.Gui.Graph.GraphCanvas.DeleteSelectedElements() in C:\dev\repos\t3\t3\Gui\Graph\GraphCanvas.cs:line 867
   at T3.Gui.Graph.GraphCanvas.Draw(ImDrawListPtr dl, Boolean showGrid) in C:\dev\repos\t3\t3\Gui\Graph\GraphCanvas.cs:line 217
   at T3.Gui.Graph.GraphWindow.DrawContent() in C:\dev\repos\t3\t3\Gui\Graph\GraphWindow.cs:line 233
   at T3.Gui.Windows.Window.DrawOneInstance() in C:\dev\repos\t3\t3\Gui\Windows\Window.cs:line 93
   at T3.Gui.Graph.GraphWindow.DrawAllInstances() in C:\dev\repos\t3\t3\Gui\Graph\GraphWindow.cs:line 127
   at T3.Gui.Windows.Window.Draw() in C:\dev\repos\t3\t3\Gui\Windows\Window.cs:line 49
   at T3.Gui.Windows.WindowManager.Draw() in C:\dev\repos\t3\t3\Gui\Windows\WindowManager.cs:line 96
   at T3.Gui.T3Ui.Draw() in C:\dev\repos\t3\t3\Gui\T3UI.cs:line 70
   at T3.Program.<>c__DisplayClass13_0.<Main>g__RenderCallback|1() in C:\dev\repos\t3\t3\Program.cs:line 225
   at SharpDX.Windows.RenderLoop.Run(Control form, RenderCallback renderCallback, Boolean useApplicationDoEvents)
   at T3.Program.Main() in C:\dev\repos\t3\t3\Program.cs:line 241

```

This is how the operator looks like 

![image](https://user-images.githubusercontent.com/9899825/193419000-df1ad461-ec37-43c5-9b9f-df533f742199.png)

x,y float, InputImage Texture2D, output Vector4

cs file is basically empty (returns only a const vec4)

**Comment by pixtur** [comment]
Interesting. Does the crash still appear after saving and restarting?
If yes, could you share the operator in question?

**Comment by hollyhook** [comment]
Crash is still reproducible, the operator is the one which is subject of https://github.com/still-scene/t3/pull/180

**Comment by pixtur** [comment]
Fixed in upcoming version 3.5
